app/services/ExactTargetService.scala

```diff
@@ -51,17 +54,17 @@ trait ExactTargetService extends LazyLogging {
       purchaserIds: PurchaserIdentifiers): Future[DataExtensionRow] = {
 
     val zuoraPaidSubscription: Future[PaidSub] =
-      subscriptionData.productData.fold(
-        { paper => paperSubscriptionService.unsafeGetPaid(Subscription.Name(subscribeResult.subscriptionName))},
-        { digipack => digiSubscriptionService.unsafeGetPaid(Subscription.Name(subscribeResult.subscriptionName))}
-      ).andThen { case Failure(e) =>
-        logger.error(s"Failed to get paid subscription ${subscribeResult.subscriptionName} from Zuora for user ${purchaserIds.identityId}", e)}
+      retry.Backoff(max = 2, delay = 2.seconds, base = 2) { () =>
+        subscriptionData.productData.fold(
+          { paper => paperSubscriptionService.unsafeGetPaid(Subscription.Name(subscribeResult.subscriptionName)) },
+          { digipack => digiSubscriptionService.unsafeGetPaid(Subscription.Name(subscribeResult.subscriptionName)) }).either
+      }.map(_.fold(e => throw new Exception(s"Failed to get Zuora paid subscription ${subscribeResult.subscriptionName}", e), identity))
 
     val zuoraPaymentMethod: Future[PaymentMethod] =
-      paymentService.getPaymentMethod(Subscription.AccountId(subscribeResult.accountId)).map(
-        _.getOrElse(throw new Exception(s"Subscription with no payment method found, ${subscribeResult.subscriptionId}"))
-      ).andThen { case Failure(e) =>
-        logger.error(s"Failed to get payment method from Zuora: subscriptionId=${subscribeResult.subscriptionName} userId=${purchaserIds.identityId}", e)}
+      retry.Backoff(max = 2, delay = 2.seconds, base = 2) { () =>
```

Comment:
Perhaps some kind of method of type ( => Future => Future) called retry would be nice, that did all the logging of failures etc would be good in the future